samples/msal-browser-samples/VanillaJSTestApp2.0/app/boundRt/test/browser.spec.ts

```diff
@@ -0,0 +1,73 @@
+import "mocha";
+import puppeteer from "puppeteer";
+import { expect } from "chai";
+import { Screenshot, createFolder, setupCredentials, enterCredentials } from "../../../../../e2eTestUtils/TestUtils";
+import { BrowserCacheUtils } from "../../../../../e2eTestUtils/BrowserCacheTestUtils";
+import { LabApiQueryParams } from "../../../../../e2eTestUtils/LabApiQueryParams";
+import { AzureEnvironments, AppTypes } from "../../../../../e2eTestUtils/Constants";
+import { LabClient } from "../../../../../e2eTestUtils/LabClient";
+
+const SCREENSHOT_BASE_FOLDER_NAME = `${__dirname}/screenshots`;
+const SAMPLE_HOME_URL = "http://localhost:30662/";
+let username = "";
+let accountPwd = "";
+
+describe("Browser tests", function () {
+    this.timeout(0);
+    this.retries(1);
+
+    let browser: puppeteer.Browser;
+    before(async () => {
+        createFolder(SCREENSHOT_BASE_FOLDER_NAME);
+        const labApiParams: LabApiQueryParams = {
+            azureEnvironment: AzureEnvironments.PPE,
+            appType: AppTypes.CLOUD
+        };
+
+        const labClient = new LabClient();
+        const envResponse = await labClient.getVarsByCloudEnvironment(labApiParams);
+        [username, accountPwd] = await setupCredentials(envResponse[0], labClient);
+        
+        browser = await puppeteer.launch({
+            headless: true,
+            ignoreDefaultArgs: ["--no-sandbox", "–disable-setuid-sandbox"]
+        });
+    });
+
+    let context: puppeteer.BrowserContext;
+    let page: puppeteer.Page;
+    let BrowserCache: BrowserCacheUtils;
+    beforeEach(async () => {
+        context = await browser.createIncognitoBrowserContext();
+        page = await context.newPage();
+        BrowserCache = new BrowserCacheUtils(page, "sessionStorage");
+        await page.goto(SAMPLE_HOME_URL);
+    });
+
+    afterEach(async () => {
+        await page.close();
+    });
+
+    after(async () => {
+        await context.close();
+        await browser.close();
+    });
+
+    it("Performs loginRedirect with RT binding", async () => {
+        const testName = "redirectBaseCase";
+        const screenshot = new Screenshot(`${SCREENSHOT_BASE_FOLDER_NAME}/${testName}`);
+        // Home Page
+        await page.waitForSelector("#SignIn");
+        await screenshot.takeScreenshot(page, "samplePageInit");
+        // Click Sign In
+        await page.click("#SignIn");
+        await page.waitForSelector("#loginRedirect");
+        await screenshot.takeScreenshot(page, "signInClicked");
+        // Click Sign In With Redirect
+        await page.click("#loginRedirect");
+        // Check URL to see if it contains stkJwk
+        await page.waitForNavigation({ waitUntil: "networkidle0", timeout: 10000});
+        await page.waitForSelector("input#i0116.input.text-box");
+        expect(page.url()).to.include("stk_jwk");
```

Comment:
Any way we can test the full E2E flow here, given that the integration tests should already cover this?

Reply:
These PRs slowly increase what is tested as the feature gets developed. The reason we can't test the whole flow is that the auth response at this point is encrypted and the decryption steps are added in subsequent PRs.

I'm relying heavily on the staging branch and the feature flag to be able to separate these PRs in a way that allow for the protocol to be reviewed "in order".